templates/etc/systemd/system/caddy.service.erb

```diff
@@ -9,19 +9,20 @@ Wants=network-online.target systemd-networkd-wait-online.service
 User=<%= @caddy_user %>
 Group=<%= @caddy_group %>
 
-Environment=HOME=/etc/ssl/caddy
+Environment=HOME=<%= @caddy_home %>
+Environment=CADDYPATH=<%= @caddy_ssl_dir %>
 
-ExecStart=<%= @install_path %>/caddy -log <%= @caddy_log_dir %>/caddy.log -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp
+ExecStart=<%= @install_path %>/caddy -log <%= @caddy_log_dir %>/caddy.log -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp -http-port=<%= @caddy_http_port %> -https-port=<%= @caddy_https_port %>
 ExecReload=/bin/kill -USR1 $MAINPID
 
 LimitNOFILE=1048576
-LimitNPROC=64
+LimitNPROC=512
```

Comment:
Isn't 64 already quite high? Maybe this should be a parameter too?

Reply:
I've changed this because it is recommended by the creator of Caddy, as he notes:

"Unmodified caddy is not expected to use more than that."

Which means it is possible that a default caddy configuration could use more than 64 processes, but less than 512. (according to mholt)

ref: https://github.com/mholt/caddy/blob/master/dist/init/linux-systemd/caddy.service